packages/browser/__tests__/login/oidc/OidcLoginHandler.spec.ts

```diff
@@ -152,6 +189,38 @@ describe("OidcLoginHandler", () => {
     expect(storedClientId).toEqual(inputClientId);
   });
 
+  it("should save client WebID if one is provided, and the target IdP supports Solid-OIDC", async () => {
+    const mockedStorage = new StorageUtility(mockStorage({}), mockStorage({}));
+
+    const actualHandler = defaultMocks.oidcHandler;
+    const handler = getInitialisedHandler({
+      oidcHandler: actualHandler,
+      storageUtility: mockedStorage,
+      clientRegistrar: new ClientRegistrar(mockedStorage),
+      issuerConfigFetcher: mockIssuerConfigFetcher({
+        ...IssuerConfigFetcherFetchConfigResponse,
+        solidOidcSupported: true,
+      }),
+    });
+
+    await handler.handle({
+      sessionId: "mySession",
+      oidcIssuer: "https://arbitrary.url",
+      redirectUrl: "https://app.com/redirect",
+      tokenType: "DPoP",
+      clientId: "https://my.app/registration",
```

Comment:
Just to emphasise that client ID's are expected to be WebID's (should app WebID's end with `#app`...??):
```suggestion
      clientId: "https://my.app/profile/card#app",
```

Reply:
I added a hash fragment in the client WebIDs in 0f5f5d58f80952d1905cb73a4e7aad9b6016e3ed, but I'd prefer to avoid associating the notion of WebID to the `/profile/card` IRI template, which is merely a convention, but isn't something that we should necessarily expect.

Reply:
Sure, the `#app` is better (since it's the fragment part that makes the IRI a non-information resource, which is what a WebID really is). Personally, I still like conforming to the current convention of `/profile/card` (and extending that convention with `#app` for client app WebID's maybe), as I think having a convention (whatever that convention might be) still helps in communicating the intent (i.e., if I see `/profile/card` on *anything*, I'll automatically assume it's a WebID), and I think there's inherent value in that. But I'm still fine with you not doing so either (since it is 'just' a convention at the moment), and I've never really liked how long adding `/profile/card` makes a WebID anyways (and needing to remember whether it's `/profile/card` or `/card/profile` !).